src/Datastore/Operation.php

```diff
@@ -405,7 +405,7 @@ public function runQuery(QueryInterface $query, array $options = [])
                     yield $result;
                 }
 
-                if ($query->canPaginate() && $res['batch']['moreResults'] !== 'NO_MORE_RESULTS') {
+                if ($query->canPaginate() && $res['batch']['moreResults'] === 'NOT_FINISHED') {
```

Comment:
What is the limit that dictates a cut off and for further batches to be returned?